pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -166,6 +166,10 @@ func (rc *reconciler) reconcile() {
 	// Ensure volumes that should be unmounted are unmounted.
 	for _, mountedVolume := range rc.actualStateOfWorld.GetMountedVolumes() {
 		if !rc.desiredStateOfWorld.PodExistsInVolume(mountedVolume.PodName, mountedVolume.VolumeName) {
+			if rc.operationExecutor.IsOperationPending(mountedVolume.VolumeName, nestedpendingoperations.EmptyUniquePodName) {
+				klog.V(5).Infof("Skipping UnmountVolume, device operation is in progress")
+				continue
```

Comment:
How is this different from the check that the executer will do internally? See here:
https://github.com/kubernetes/kubernetes/blob/b7e2980a57289a32a7dbc5165584a0b1a56c9dd9/pkg/volume/util/nestedpendingoperations/nestedpendingoperations.go#L94-L108

When I asked about it in https://github.com/kubernetes/kubernetes/issues/70319 and later in a CSI WG meeting, @saad-ali said that the executioner will prevent running multiple operations against the same volume in parallel.

It wasn't obvious how that works and I was still worried that the executioner would merely serialize operations. But after looking at the code above, my interpretation is that it will discard new operations *instead* of queuing them, so it should do something similar to what you propose here.

But I am not entirely sure - perhaps your check is stricter, or I misunderstood something.

Reply:
The difference is `podName` - (Un)MountDevice operation uses empty pod name, it's global for all pods, while (Un)MountVolume uses pod names. IMO, (volumeName, podName) is the primary key in the operation map, so multiple MountVolumes can run in parallel on the same volume.